### DISCUSSION TITLE
Manual installation: Fatal error: PyQt5.QtCore is required

### DISCUSSION THREAD

**Discussion by gadefox** [qutebrowser/qutebrowser#6656] [Q&A] [answered]
I know this is not a bug, because I can launch the app when all python wrappers are installed. Now I want to install only qutebrowser dependencies, but I'm getting following error. Installed Qt5 packages are listed at the bottom. What am I missing?

(venv) fox@debian:~/.build/adblock-0.4.4$ qutebrowser 
Fatal error: PyQt5.QtCore is required to run qutebrowser but could not be imported! Maybe it's not installed?

The error encountered was: No module named 'PyQt5'

-------------------------------------------------
python3-pyqt5.qtopengl/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtpositioning/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtquick/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtsensors/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtsql/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtsvg/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtwebchannel/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.qtwebengine/stable,now 5.15.2-2 i386 [installed]
python3-pyqt5.qtx11extras/stable,now 5.15.2+dfsg-3 i386 [installed]
python3-pyqt5.sip/stable,now 12.8.1-1+b2 i386 [installed]
python3-pyqt5/stable,now 5.15.2+dfsg-3 i386 [installed]

libqt5concurrent5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5core5a/stable,now 5.15.2+dfsg-9 i386 [installed]
libqt5dbus5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5designer5/stable,now 5.15.2-5 i386 [installed,automatic]
libqt5designercomponents5/stable,now 5.15.2-5 i386 [installed,automatic]
libqt5gui5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5help5/stable,now 5.15.2-5 i386 [installed]
libqt5location5/stable,now 5.15.2+dfsg-2 i386 [installed,automatic]
libqt5multimedia5/stable,now 5.15.2-3 i386 [installed,automatic]
libqt5network5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5opengl5-dev/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5opengl5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5positioning5/stable,now 5.15.2+dfsg-2 i386 [installed,automatic]
libqt5positioningquick5/stable,now 5.15.2+dfsg-2 i386 [installed,automatic]
libqt5printsupport5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5qml5/stable,now 5.15.2+dfsg-6 i386 [installed,automatic]
libqt5qmlmodels5/stable,now 5.15.2+dfsg-6 i386 [installed,automatic]
libqt5quick5/stable,now 5.15.2+dfsg-6 i386 [installed,automatic]
libqt5quickwidgets5/stable,now 5.15.2+dfsg-6 i386 [installed,automatic]
libqt5sensors5/stable,now 5.15.2-2 i386 [installed,automatic]
libqt5sql5-sqlite/stable,now 5.15.2+dfsg-9 i386 [installed]
libqt5sql5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5svg5/stable,now 5.15.2-3 i386 [installed,automatic]
libqt5test5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5waylandclient5/stable,now 5.15.2-3 i386 [installed,automatic]
libqt5waylandcompositor5/stable,now 5.15.2-3 i386 [installed,automatic]
libqt5webchannel5/stable,now 5.15.2-2 i386 [installed,automatic]
libqt5webengine-data/stable,now 5.15.2+dfsg-3 all [installed,automatic]
libqt5webengine5/stable,now 5.15.2+dfsg-3 i386 [installed]
libqt5webenginecore5/stable,now 5.15.2+dfsg-3 i386 [installed]
libqt5webenginewidgets5/stable,now 5.15.2+dfsg-3 i386 [installed,automatic]
libqt5webkit5/stable,now 5.212.0~alpha4-11 i386 [installed,automatic]
libqt5widgets5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
libqt5x11extras5/stable,now 5.15.2-2 i386 [installed,automatic]
libqt5xml5/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
qdoc-qt5/stable,now 5.15.2-5 i386 [installed,automatic]
qhelpgenerator-qt5/stable,now 5.15.2-5 i386 [installed,automatic]
qt5-assistant/stable,now 5.15.2-5 i386 [installed,automatic]
qt5-qmake-bin/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
qt5-qmake/stable,now 5.15.2+dfsg-9 i386 [installed]
qtattributionsscanner-qt5/stable,now 5.15.2-5 i386 [installed,automatic]
qtbase5-dev-tools/stable,now 5.15.2+dfsg-9 i386 [installed,automatic]
qtbase5-dev/stable,now 5.15.2+dfsg-9 i386 [installed]
qttools5-dev-tools/stable,now 5.15.2-5 i386 [installed,automatic]
qttools5-dev/stable,now 5.15.2-5 i386 [installed]
qttranslations5-l10n/stable,now 5.15.2-2 all [installed,automatic]
qtwayland5/stable,now 5.15.2-3 i386 [installed,automatic]

**Comment by The-Compiler** [comment, marked as answer]
> I know this is not a bug, because I can launch the app when all python wrappers are installed. 

I thus converted it into a discussion.

> Installed Qt5 packages are listed at the bottom. What am I missing?

You'll need `PyQt5.QtCore` as the error message says. In your case, that'd be `python3-pyqt5` - however, according to the prompt, you're running from a virtualenv, so the system-wide packages are probably not relevant at all.

**Reply by gadefox** [reply]
Oh my.. anyway.. Symlink is the solution in this case:
`sudo ln -s /usr/lib/python3/dist-packages/PyQt5-5.15.2.dist-info /usr/local/venv/qutebrowser/lib/python3.9/site-packages/PyQt5-5.15.2.dist-info; sudo ln -s /usr/lib/python3/dist-packages/PyQt5_sip-12.8.1.egg-info /usr/local/venv/qutebrowser/lib/python3.9/site-packages/PyQt5_sip-12.8.1.egg-info; sudo ln -s /usr/lib/python3/dist-packages/PyQtWebEngine-5.15.2.dist-info /usr/local/venv/qutebrowser/lib/python3.9/site-packages/PyQtWebEngine-5.15.2.dist-info; sudo ln -s /usr/lib/python3/dist-packages/PyQt5 /usr/local/venv/qutebrowser/lib/python3.9/site-packages/PyQt5
`

Re "I can launch the app when all python wrappers are installed": It has nothing to do with this. I just don't use python virtual env on that pc..
